presto-common/src/main/java/com/facebook/presto/common/Page.java

```diff
@@ -329,6 +329,11 @@ public Page copyPositions(int[] retainedPositions, int offset, int length)
         return wrapBlocksWithoutCopy(length, blocks);
     }
 
+    public Page extractChannel(int channel)
+    {
+        return wrapBlocksWithoutCopy(positionCount, new Block[]{this.blocks[channel]});
```

Comment:
Will this save CPU c.p. `extractChannels(new int[]{channel});`?

Reply:
They should be the same IFF the method is fully inlined and the temporary int array becomes stack allocated, but that seemed more fragile to depend on compared to just providing the desired implementation directly.